modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/persistence/checkpoint/Checkpointer.java

```diff
@@ -634,10 +634,9 @@ public void cancel() {
             log.debug("Cancelling grid runnable: " + this);
         }
 
-        // Do not interrupt runner thread.
-        isCancelled.set(true);
-
         synchronized (this) {
+            // Do not interrupt runner thread.
+            isCancelled.set(true);
```

Comment:
You need to return it back, taking the monitor may take longer than it seems at first glance, and `isCancelled` is also read without a monitor.